src/pages/instances/InstanceSearchFilter.tsx

```diff
@@ -91,6 +96,24 @@ const InstanceSearchFilter: FC<Props> = ({ instances, setFilters }) => {
         }
         filterPanelData={searchAndFilterData}
         returnSearchData={onSearchDataChange}
+        onExpandChange={() => {
+          window.dispatchEvent(
+            new CustomEvent("resize", { detail: "search-and-filter" })
+          );
+        }}
+        onPanelToggle={() => {
+          const searchAndFilterPanel = document.querySelector(
+            ".p-search-and-filter__panel"
+          );
+          if (!searchAndFilterPanel) {
+            return;
+          }
+          const isShowing =
+            searchAndFilterPanel.getAttribute("aria-hidden") === "false";
+          if (!menuCollapsed && isShowing && isSmallScreen()) {
+            setMenuCollapsed(true);
+          }
```

Comment:
I think we can reduce the coupling between components here.

`onPanelToggle` will throw a custom event, similar to the one `onExpandChange` produces. In `menuCollapsed.tsx` we listen to that event and keep all the logic isSmallScreen, and the menuCollapsed / setMenuCollapsed local to it. We can then use that hook exclusively within Navigation and avoid the need for the `MenuCollapsedProvider`, because using the hook in only one place which has the state.